src/util/piped_process.cpp

```diff
@@ -30,8 +101,127 @@
 
 piped_processt::piped_processt(const std::vector<std::string> commandvec)
 {
+  // Default state
+  process_state = statet::NOT_CREATED;
 #  ifdef _WIN32
-  UNIMPLEMENTED_FEATURE("Pipe IPC on windows.")
+  // Security attributes for pipe creation
+  SECURITY_ATTRIBUTES sec_attr;
+  // Ensure pipes are inherited
+  sec_attr.nLength = sizeof(SECURITY_ATTRIBUTES);
+  sec_attr.bInheritHandle = TRUE;
+  // This sets the security to the default for the current session access token
+  // See following link for details
+  // https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/aa379560(v=vs.85) //NOLINT
+  sec_attr.lpSecurityDescriptor = NULL;
+  // Use named pipes to allow non-blocking read
+  // Build the base name for the pipes
+  std::string base_name = "\\\\.\\pipe\\cbmc\\SMT2\\child\\";
+  // Use process ID as a unique ID for this process at this time.
+  base_name.append(std::to_string(GetCurrentProcessId()));
+  std::string tmp_name = base_name;
+  tmp_name.append("\\IN");
+  LPCSTR tmp_str = tmp_name.c_str();
+  child_std_IN_Rd = CreateNamedPipe(
+    tmp_str,
+    PIPE_ACCESS_INBOUND,          // Reading for us
+    PIPE_TYPE_BYTE | PIPE_NOWAIT, // Bytes and non-blocking
+    PIPE_UNLIMITED_INSTANCES,     // Probably doesn't matter
+    BUFSIZE,
+    BUFSIZE,    // Output and input bufffer sizes
+    0,          // Timeout in ms, 0 = use system default
```

Comment:
This is the timeout to detect that the other side hasn't connected, I guess?

Reply:
Any insights on this? Would be good to explain to save follow-ups the time for investigating how the API works.